oceanspy/utils.py

```diff
@@ -18,6 +18,26 @@
 except ImportError:  # pragma: no cover
     pass
 
+# Jiang being a bit lazy
+import numpy as np
+
+try:
+    from numba import njit
+except ImportError:  # pragma: no cover
+
+    def njit(f):
+        return f
```

Comment:
I think it would be a bit cleaner to do something like this:
```suggestion
import numpy as np

try:
    import numba
    has_numba=True
except ImportError:
    has_numba=False

def conditional_njit(f):
    if has_numba:
        return numba.njit(f)
    return f
```

I imagine you are seeing quite a bit of improvement using numba. Fell free to use the decorator on other functions as well!

Reply:
Thanks a lot for the comment! I assume I need to change all the @njit to @conditional_njit in the rest of the code. Is that correct?

Reply:
correct! also, I haven't tested the snippet above - hopefully it works fine

Reply:
No worries. I am going to test it